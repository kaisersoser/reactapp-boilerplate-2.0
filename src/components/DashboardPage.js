import React from 'react';

export const DashboardPage = () => (
  <div>
    Dashboard Page content
  </div>
);

export default DashboardPage;
