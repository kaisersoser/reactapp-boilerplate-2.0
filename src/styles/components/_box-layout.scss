.box-layout {
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out($color: #ffffff, $amount: .35);
    border-radius: 3px;
    box-shadow: 5px 5px 2px rgba($color: dimgray, $alpha: 0.55);
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;    
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}