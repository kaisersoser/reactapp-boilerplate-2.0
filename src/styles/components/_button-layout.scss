.button-layout {
    align-items: center;
    background: $button-blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $button-font-size;  
    font-weight: 300;  
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button-layout--link {
    background: none;
}

.button-layout--secondary {
    background: #888888;
}