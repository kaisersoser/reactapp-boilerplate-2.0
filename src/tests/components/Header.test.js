import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let startLogout, wrapper;

beforeEach(() => {    
    startLogout = jest.fn();
    wrapper = shallow(<Header  startLogout={startLogout}/>);
});

test('Should test Header is properly rendered', () => {
    const wrapper = shallow(<Header startLogout={() => {}} />);
    expect(wrapper).toMatchSnapshot();
});


test('Should test the start logout click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});
