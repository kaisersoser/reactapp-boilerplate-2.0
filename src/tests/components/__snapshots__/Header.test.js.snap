// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test Header is properly rendered 1`] = `
<header
  className="header-layout"
>
  <div
    className="content-container"
  >
    <div
      className="header-layout__content"
    >
      <Link
        className="header-layout__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate v2.0
        </h1>
      </Link>
      <button
        className="button-layout button-layout--link"
        onClick={[Function]}
      >
        Sign out
      </button>
    </div>
  </div>
</header>
`;
